src/wp-includes/global-styles-and-settings.php

```diff
@@ -336,22 +375,10 @@ function wp_add_global_styles_for_blocks() {
 				wp_add_inline_style( $stylesheet_handle, $block_css );
 			}
 		}
+	}
 
-		// The likes of block element styles from theme.json do not have  $metadata['name'] set.
```

Comment:
After doing some more debugging. I believe this change, along with trying to resolve the block name cache key using the `$metadata['path']` property is what is causing the remaining test failures here. In this particular case, there are styles included in the theme.json data for the `my/third-party-block` block followed by custom styles targeting the `cite` element inside that block as separate block nodes. When the path is used to resolve the block name used as the key, the first node gets added to the cache, causing the second one to be skipped.

I'll open a separate PR with an example of how to fix this, but it means that only nodes with name keys would get cached. In retrospect, maybe your original idea of using a hash of each node as the key was a better solution as long as it doesn't cause a large performance penalty.